File: 2023/czerwiec/02/rozwiazanie/konsola/sortowanie.java
import java.util.Random;

public class sortowanie
{
    public static void fill(int[] arr, int max)
    {
        Random rnd = new Random();
        for (int i = 0; i < arr.length; i++)
        {
            arr[i] = rnd.nextInt(max + 1);
        }
    }
    public static void sort(int[] arr)
    {
        for(int i = 0; i < arr.length; i++)
        {
            boolean flag = false;
            for(int j = 0; j < arr.length-i-1; j++)
            {
                if (arr[j] > arr[j+1])
                {
                    int buff = arr[j+1];
                    arr[j+1] = arr[j];
                    arr[j]=buff;
                    flag = true;
                }
            }
            if (!flag)
            {
                break;
            }
        }
    }
    public static void print(int[] arr)
    {
        System.out.println("Wynik sortowania bąbelkowego:");
        System.out.print(arr[0]);
        for (int i = 1; i < arr.length; i++)
        {
            System.out.print(", " + arr[i]);
        }
    }
    public static void main(String[] args)
    {
        int[] nums = new int[100]; 

        fill(nums, 100);
        sort(nums);
        print(nums);
    }
}

File: 2023/czerwiec/02/rozwiazanie/konsola/sortowanie.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

const int nums_count = 100;

void fill(int* arr, int max = 100)
{
    srand(time(NULL));
    for (int i = 0; i < nums_count; ++i)
    {
        arr[i] = rand() % (max + 1);
    }
}
void sort(int* arr)
{
    bool flag;
    int buff;

    for (int i = 0; i < nums_count; ++i)
    {
        flag = false;
        for (int j = 0; j < nums_count - i - 1; j++)
        {
            if (arr[j] > arr[j + 1])
            {
                buff = arr[j + 1];
                arr[j + 1] = arr[j];
                arr[j] = buff;
            }
        }
    }
}
void print(int* nums)
{
    std::cout << "Wynik sortowania bąbelkowego:\n";
    std::cout << nums[0];
    for (int i = 1; i < nums_count; i++)
    {
        std::cout << ", " << nums[i];
    }
    std::cout << std::endl;
}

int main()
{
    int nums[nums_count];

    fill(nums, 100);
    sort(nums);
    print(nums);

    return 0;
}